clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -231,6 +243,17 @@ private void handleCoordinatorDisconnect(Throwable exception, long currentTimeMs
         }
     }
 
+    @Override
+    public NetworkClientDelegate.PollResult pollOnClose() {
+        if (!pendingRequests.hasUnsentRequests() || !coordinatorRequestManager.coordinator().isPresent())
+            return EMPTY;
+
+        sendAutoCommit(subscriptions.allConsumed());
```

Comment:
We seem to have two code paths here dealing with the auto commit and I haven't quite grasped what is the idea here. For example, we seem to be creating the auto-commit request twice during close:

```
ConsumerNetworkThread.coordinatorOnClose -> ConsumerNetworkThread.closingTasks -> ConsumerNetworkThread.maybeAutoCommitOnClose -> CommitRequestManager.maybeAutoCommitOnClose -> CommitRequestManager.createOffsetCommitRequest
```

As well as:

```
ConsumerNetworkThread.runAtClose -> CommitRequestManager.pollOnClose -> CommitRequestManager.sendAutoCommit -> CommitRequestManager.addOffsetCommitRequest -> CommitRequestManager.creatOffsetCommitRequest
```

Could you add a bit of context to the PR so that it's easier for me to understand what is the thinking behind the shutdown procedure?